python/tach/sync.py

```diff
@@ -84,12 +84,26 @@ def sync_project(
             "Unexpected error. Could not find configuration file during 'sync'."
         )
 
-    project_config = sync_dependency_constraints(
+    if not add:
+        existing_modules: list[ModuleConfig] = []
+        for module in project_config.modules:
+            module_path = fs.module_to_pyfile_or_dir_path(
+                tuple(project_config.source_roots), module.path
+            )
+            if module_path is not None:
+                existing_modules.append(module)
+            else:
+                print(f"Removing non-existent module: {module.path}")
+        project_config = ProjectConfig(
+            **project_config.model_dump(exclude={"modules"}), modules=existing_modules
+        )
```

Comment:
I think it is better to combine this logic with the section on line 31 within sync_dependency_constraints. Using `model_copy` will be faster than dumping and revalidating, and it is easier to reason about the code when all of the prune-related config changes are next to each other.

Reply:
@emdoyle I made the changes but that caused two `check` tests to fail as check compares the synced modules with the config and throws error for unused dependencies.

Reply:
Any thoughts. why that would happen?
I am not able to understand

Reply:
<img width="461" alt="image" src="https://github.com/user-attachments/assets/2d9b435a-fe50-41e9-afac-35c4eaa42fca">

When I run `tach sync` and `tach check` inside valid example project. No erros are shown.

Reply:
Thanks for the details, I pulled down the branch locally and was able to debug. Essentially all modules were being pruned out because `source_roots` is expected as absolute paths, meaning with relative paths from the test directory, none of the modules were being found. That also would explain why running it manually from the directory itself hides the issue :)

I made a small change so that the source roots are absolute and now tests are passing as expected.

Reply:
Got it.